hs/CourseographyFacebook.hs

```diff
@@ -85,23 +97,24 @@ performFBAction action = withManager $ \manager -> FB.runFacebookT credentials m
 -- | Posts a message to the user's Facebook feed, with the code 'code'. GraphResponse
 -- is then sent back to the user.
 postToFacebook :: String -> ServerPart Response
-postToFacebook code = (liftIO $ performPost code) >> graphResponse
+postToFacebook code = (liftIO $ performPost (BS.pack code)) >> graphResponse
 
 -- | Performs the posting to facebook.
-performPost :: String -> IO Response
+performPost :: BS.ByteString -> IO Response
 performPost code = 
-	performFBAction $ do
-        postToFB code =<< getToken testPostUrl code
+    performFBAction $ do
+        token <- getToken testPostUrl code
+        user <- FB.getUser "me" [] (Just token)
+        let id_ = FB.userId user
+        liftIO $ createPNGFile ("INSERT_ID" ++ "-graph.png")
```

Comment:
In the above file, you hard-coded this. I'd be happy doing the same here. (For now, of course.)